src/Microsoft.DotNet.CodeFormatting.Tests/PropertyBagTests.cs

```diff
@@ -0,0 +1,441 @@
+﻿// /********************************************************
+// *                                                       *
+// *   Copyright (C) Microsoft. All rights reserved.       *
+// *                                                       *
+// ********************************************************/
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Binary;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.CodeAnalysis.Options
+{
+    [TestClass]
+    public class PropertyBagTests
+    {
+        [TestMethod]
+        public void PropertyBag_TryConvertFromString()
+        {
+            var propertyBag = new PropertyBag();
+            propertyBag["TestValue"] = "15";
+            int converted;
+            Assert.IsTrue(propertyBag.TryGetProperty("TestValue", out converted));
+            Assert.AreEqual(15, converted, "String conversion to primitive type did not succeed.");
+        }
+
+        [TestMethod]
+        public void PropertyBag_NoCache()
+        {
+            var propertyBag = new PropertyBag();
+
+            TestEnum testEnum = propertyBag.GetProperty(TestEnumOptionTwo, cacheDefault: false);
+
+            Assert.AreEqual(TestEnumOptionTwo.DefaultValue, testEnum);
+            Assert.AreEqual(propertyBag.Count, 0);
+        }
+
+        [TestMethod]
+        public void PropertyBag_TypedPropertyBagNoCache()
+        {
+            var propertyBag = new TestEnumPropertyBag();
+
+            TestEnum testEnum = propertyBag.GetProperty(TestEnumOptionTwo, cacheDefault: false);
+
+            Assert.AreEqual(TestEnumOptionTwo.DefaultValue, testEnum);
+            Assert.AreEqual(propertyBag.Count, 0);
+        }
+
+        [TestMethod]
+        public void PropertyBag_InitializeFromPropertyBag()
+        {
+            int expectedValue = 72;
+            var propertyBag = new PropertyBag();
+            propertyBag["TestValue"] = expectedValue.ToString();
+            
+            var copiedPropertyBag = new PropertyBag(propertyBag, StringComparer.OrdinalIgnoreCase);
+
+            int converted;
+            Assert.IsTrue(propertyBag.TryGetProperty("TestValue", out converted));
+            Assert.AreEqual(expectedValue, converted, "String conversion to primitive type did not succeed.");
+
+            Assert.IsFalse(propertyBag.TryGetProperty("testvalue", out converted));
+
+            Assert.IsTrue(copiedPropertyBag.TryGetProperty("TestValue", out converted));
+            Assert.AreEqual(expectedValue, converted, "String conversion to primitive type did not succeed.");
```

Comment:
Check that the `"TestValue"` property in the `copiedPropertyBag` actually has the value "72".